calculator.js

```diff
@@ -0,0 +1,47 @@
+const operators = ['+', '-', '*', '/'];
+
+exports.create_calculator = function (operator) {
+  const numbers = [];
+
+  return {
+    add_number: function (number) {
+      numbers.push(number);
+    },
+    calculate: function () {
+      // Return zero if there are no numbers to calculate
+      if (numbers.length === 0) {
+        throw 'Cannot calculate a result because no numbers were given';
+      }
+
+      let answer = numbers[0];
+
+      let i = 0;
+      numbers.forEach(function (number) {
+        if (i === 0) {
```

Comment:
Can you please explain what i is doing here.. it doesn't seem to be used anywhere in the forEach?

Reply:
Aah I see what you are doing.. we are just using i to skip the first number in the loop.